### PR TITLE
Toggle to disable the pLS duplicate cleaning

### PR DESCRIPTION
**Pull request by VourMa**
This PR introduces a toggle (-2) to disable pLS duplicate cleaning. This is not meant to be a proposed change, as it both increases the timing and has dramatic physics performance changes, but it serves as a technical change to facilitate more configuration when including Patatrack pixel tracks as input in the future.

This is the master version of the now closed PR #318, and is only possible for merging after the updates in PR #324.

### Timing
**Before** (48e470e0b04c45eb66a33ec7874e27670fe00473)
```
   Evt    Hits       MD       LS      T3       T5       pLS       pT5      pT3      TC       Reset    Event     Short             Rate
   avg      2.1      1.4      0.5      2.5      2.9      0.6      1.4      1.1      1.6      0.4      14.7      12.0+/-  2.6      16.1   explicit_cache[s=1]
   avg      3.2      1.7      0.7      3.8      3.8      0.7      2.2      1.6      2.3      0.7      20.8      16.9+/-  3.7      11.2   explicit_cache[s=2]
   avg      6.2      2.3      1.4      6.3      6.2      0.9      4.3      2.8      3.9      1.1      35.4      28.4+/-  5.3       9.3   explicit_cache[s=4]
   avg     10.0      3.6      2.2      8.4      8.9      1.1      6.9      3.8      5.2      1.4      51.7      40.5+/-  8.2       9.0   explicit_cache[s=6]
   avg     15.3      4.2      2.5     10.2     10.7      1.3      8.6      5.1      6.5      1.5      66.1      49.5+/- 11.5       8.6   explicit_cache[s=8]
```

**After** (4d2963adaa2db1d226c4efe01be5d217f5106eea)
```
   Evt    Hits       MD       LS      T3       T5       pLS       pT5      pT3      TC       Reset    Event     Short             Rate
   avg      2.0      1.4      0.5      2.6      2.9      0.6      1.4      1.1      1.6      0.4      14.6      12.0+/-  2.6      15.9   explicit_cache[s=1]
   avg      3.2      1.8      0.8      3.9      3.9      0.7      2.3      1.7      2.4      0.8      21.3      17.4+/-  4.0      11.5   explicit_cache[s=2]
   avg      6.3      2.4      1.4      6.1      6.1      0.9      4.4      2.7      3.7      1.3      35.2      28.0+/-  5.7       9.4   explicit_cache[s=4]
   avg     12.7      3.4      2.0      7.8      8.4      1.1      6.2      3.7      5.4      1.7      52.5      38.7+/- 11.8       9.3   explicit_cache[s=6]
   avg     15.2      4.4      2.7     10.7     11.2      1.5      9.1      5.3      6.5      1.7      68.2      51.5+/- 12.2       8.8   explicit_cache[s=8]
```

**After** (4d2963adaa2db1d226c4efe01be5d217f5106eea) - **No pLS duplicate cleaning**
```
   Evt    Hits       MD       LS      T3       T5       pLS       pT5      pT3      TC       Reset    Event     Short             Rate
   avg      2.0      1.4      0.5      2.5      2.9      0.0      9.1      1.4      1.6      0.5      21.9      19.9+/-  4.9      23.3   explicit_cache[s=1]
   avg      3.3      1.8      0.9      3.9      4.2      0.0     11.8      2.6      3.2      0.7      32.4      29.1+/-  7.6      16.9   explicit_cache[s=2]
   avg      7.1      2.8      2.1      6.1      6.2      0.0     15.9      4.3      6.6      1.0      52.2      45.0+/- 11.7      13.6   explicit_cache[s=4]
   avg     12.2      3.6      2.5      8.6      9.7      0.0     22.1      6.4      8.0      1.4      74.5      62.3+/- 17.7      13.3   explicit_cache[s=6]
   avg     19.5      5.3      3.4     11.2     12.8      0.0     24.8      8.7     10.7      1.8      98.2      78.7+/- 22.8      13.1   explicit_cache[s=8]
```

- Increase in timing anticipated by the increase of duplicates (seen below).

### Physics validation
**Before** (48e470e0b04c45eb66a33ec7874e27670fe00473) - **After** (4d2963adaa2db1d226c4efe01be5d217f5106eea)
https://uaf-10.t2.ucsd.edu/~evourlio/SDL/PR334/comparePR334_48e470-PU200_48e470D-PU200/
- No performance changes

**Before** (48e470e0b04c45eb66a33ec7874e27670fe00473) - **After** (4d2963adaa2db1d226c4efe01be5d217f5106eea) - **No pLS duplicate cleaning**
https://uaf-10.t2.ucsd.edu/~evourlio/SDL/PR334/comparePR334NopLSDupClean_48e470-PU200_48e470D-PU200/
- Efficiency seems ~unchanged:
TC_base_0_0_eff_pt
TC_base_0_0_eff_etacoarse

- Fake rate largely decreased (I don't fully understand it):
TC_fakerate_pt
TC_fakerate_etazoom

- Duplicate rate dramatically increased (as expected, when using the legacy pixel tracks as inputs, as we do currently):
TC_duplrate_pt
TC_duplrate_eta

- In general, we seem to be exchanging fake rate for duplicate rate. Again, this not meant to be configuration to work with for now.

### CODE REVIEW

File: bin/sdl_make_tracklooper [hunk 1]
```diff
@@ -18,21 +18,22 @@ usage()
...
+  echo "  -h    Help                      (Display this message)"
+  echo "  -c    cache                     (Make library with cache enabled)"
+  echo "  -s    show log                  (Full compilation script to stdout)"
+  echo "  -m    make clean binaries       (Make clean binaries before remake. e.g. when header files changed in SDL/*.h)"
+  echo "  -d    cut value ntuple          (With extra variables in a debug ntuple file)"
+  echo "  -p    primitive object ntuple   (With extra variables related to primitive objects)"
+  echo "  -3    do T3T3 extensions        (-e turned on if not specified)"
+  echo "  -N    neural networks           (Toggle LST neural networks)"
+  echo "  -C    cpu backend               (Run code on CPU, serially)"
+  echo "  -2    no pLS duplicate cleaning (Toggle LST neural networks)"
```
Comment (GNiendorf) [inline bin/sdl_make_tracklooper:30]: Think the stuff in parentheses was copied over incorrectly for the pLS flag.

Comment (VourMa) [reply bin/sdl_make_tracklooper]: Oops, right, thanks for spotting it. Fixed in [4203e70](https://github.com/SegmentLinking/TrackLooper/pull/335/commits/4203e70e7a0e39daf72f5069c1793c8daf19ee66).

**Review by GNiendorf — APPROVED** [review]
I'm approving this PR but I will let @ariostas merge it in if he has any further comments or questions.

**Review by ariostas — APPROVED** [review]
Looks good to me too. I'll merge it.

**Pull request merged**